src/Models/Media.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace RonasIT\Media\Models;
+
+use App\Models\User;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Support\Facades\Auth;
+use RonasIT\Support\Traits\ModelTrait;
+
+class Media extends Model
+{
+    use ModelTrait;
+    use SoftDeletes;
+    use HasFactory;
+
+    protected $fillable = [
+        'link',
+        'name',
+        'owner_id',
+        'is_public',
+        'meta'
+    ];
+
+    protected $casts = [
+        'is_public' => 'boolean',
+        'deleted_at' => 'date',
+        'meta' => 'array',
+    ];
+
+    protected $hidden = ['pivot'];
+
+    public function scopeApplyMediaPermissionRestrictions(Builder $query): void
+    {
+        if (!Auth::check()) {
+            $query->where('is_public', true);
+
+            return;
+        }
+
+        $user = Auth::user();
+
+        if (!$user->isAdmin()) {
+            $query->where(function ($subQuery) use ($user) {
+                $subQuery
+                    ->where('is_public', true)
+                    ->orWhere('owner_id', $user->id);
+            });
+        }
+    }
+
+    public function owner(): BelongsTo
```

Comment:
user model should be configurable https://github.com/RonasIT/laravel-chat/blob/4a45c8b9d6f49d80f8688db85ab9d1bcb4adbec6/src/Models/Message.php#L34

Reply:
Done.